plugins/renepay/uncertainty_network.c

```diff
@@ -170,15 +170,29 @@ void uncertainty_network_update(
 
 			if(!gossmap_chan_get_capacity(gossmap,chan,&cap))
 			{
-				plugin_err(pay_plugin->plugin,"%s (line %d) unable to fetch channel capacity",
-					__PRETTY_FUNCTION__,
-					__LINE__);
+				/* This can happen transiently if gossipd is
+				 * writing it to the store right now.  Set
+				 * it to larger htlc_max */
+				cap_msat = AMOUNT_MSAT(0);
+				for (int dir = 0; dir < 2; dir++)
+				{
+					struct amount_msat htlc_max;
+					if (!gossmap_chan_set(chan, dir))
+						continue;
+					htlc_max = amount_msat(fp16_to_u64(chan->half[dir].htlc_max));
+					if (amount_msat_greater(htlc_max, cap_msat))
+						cap_msat = htlc_max;
+				}
+				plugin_log(pay_plugin->plugin, LOG_UNUSUAL,
+					   "Cannot fetch capacity for channel %s: using %s",
+					   fmt_short_channel_id(tmpctx, scid),
+					   fmt_amount_msat(tmpctx, cap_msat));
```

Comment:
An easier solution would be to ignore those channels temporarily, they can still be picked up next time `uncertainty_network_update` is called again.
```suggestion
				/* This can happen transiently if gossipd is
				 * writing it to the store right now.
				 * Ignore this channel for the moment, we can
				 * pick it up next time. */
				plugin_log(pay_plugin->plugin, LOG_UNUSUAL,
					   "Cannot fetch capacity for channel %s: skipping it.",
					   fmt_short_channel_id(tmpctx, scid));
```
The only issue I can think of right now, is that we will have some channels in `gossmap` that have no counterpart in `chan_extra_map` and we would have to handle this case when the MCF graph is built (`init_linear_network` in `mcf.c` assumes that).